ribs/schedulers/_bandit_scheduler.py

```diff
@@ -0,0 +1,325 @@
+"""Provides the Bandit Scheduler."""
+import numpy as np
+
+
+class BanditScheduler:
+    """Schedules emitters with a bandit algorithm.
+
+    This implementation is based on `Cully 2021
+    <https://arxiv.org/pdf/2007.05352.pdf>`_.
+
+    .. note::
+        This class follows the similar ask-tell framework as
+        :class:`Scheduler`, and enforces similar constraints in the arguments
+        and methods. Please refer to the documentation of :class:`Scheduler`
+        for more details.
+
+    To initialize this class, first create an archive and a list of emitters
+    for the QD algorithm. The BanditScheduler will schedule the emitters using
+    the Upper Confidence Bound - 1 algorithm (UCB1). Everytime :meth:`ask` is
+    called, the emitters are sorted based on the potential reward function from
+    UCB1. Then, the top `num_active` emitters are used for ask-tell.
+
+    Args:
+        archive (ribs.archives.ArchiveBase): An archive object, e.g. one
+            selected from :mod:`ribs.archives`.
+        emitter_pool (list of ribs.archives.EmitterBase): A pool of emitters to
```

Comment:
Do we need to specify that emitters should be ordered in such a way that the first X will be selected initially?